### PR TITLE
Adding Zstream Automation Jenkinsfiles

### PR DESCRIPTION
**Pull request by varad-ahirwadkar**
This PR include Jenkins files for zstream deployment automation:
- zstream-trigger-job
- zstream-ocp4x-p8-min-direct-deploy
- zstream-ocp4x-p8-min-current-upgrade
- zstream-ocp4x-p8-min-next-upgrade

### CODE REVIEW

File: jobs/pipelines/powervm/zstreams/zstream-ocp4x-p8-min-next-upgrade/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+                    }
+                }
+            }
+        }
+    }
+    post {
+        always {
+            script {
+                try{
+                    archiveAllArtifacts("deploy/conformance-parallel-out.txt.tar.gz", "deploy/summary.txt", "deploy/vars.tfvars",
```
Comment (Shilpi-Das1) [inline jobs/pipelines/powervm/zstreams/zstream-ocp4x-p8-min-next-upgrade/Jenkinsfile:268]: @varad-ahirwadkar Was there any failure in always block as I think Try catch is present in all methods so I don't think try catch is required here

File: jobs/pipelines/powervm/zstreams/zstream-trigger-job/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+                        }
+                    }      
+                }
+            }
+        }
+    }
+
+    post {
+        always {
+            archiveArtifacts allowEmptyArchive: true, artifacts: "versions/*", fingerprint: true, onlyIfSuccessful: false
```
Comment (Shilpi-Das1) [inline jobs/pipelines/powervm/zstreams/zstream-trigger-job/Jenkinsfile:56]: @varad-ahirwadkar  Can we use here archiveAllArtifacts method?

File: jobs/pipelines/powervm/zstreams/zstream-trigger-job/Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,186 @@
+pipeline {
+    agent {
+        docker {
+            image 'quay.io/powercloud/inbound-agent:4.6-1.3'
+            args '-v /etc/resolv.conf:/etc/resolv.conf -v /var/run/docker.sock:/var/run/docker.sock'
```
Comment (Shilpi-Das1) [inline jobs/pipelines/powervm/zstreams/zstream-trigger-job/Jenkinsfile:5]: @varad-ahirwadkar   you can remove ` -v /var/run/docker.sock:/var/run/docker.sock` as this file is not using docker

File: jobs/pipelines/powervm/zstreams/zstream-ocp4x-p8-min-next-upgrade/Jenkinsfile (hunk 1 above)
Comment (varad-ahirwadkar) [reply jobs/pipelines/powervm/zstreams/zstream-ocp4x-p8-min-next-upgrade/Jenkinsfile:268]: yes sometimes it fails and skips the cleanup, so I have added try catch to ensure running cleanWs(). I will add it in finally block().

File: jobs/pipelines/powervm/zstreams/zstream-trigger-job/Jenkinsfile (hunk 2 above)
Comment (varad-ahirwadkar) [reply jobs/pipelines/powervm/zstreams/zstream-trigger-job/Jenkinsfile:56]: yes sure

File: jobs/pipelines/powervm/zstreams/zstream-trigger-job/Jenkinsfile (hunk 3 above)
Comment (varad-ahirwadkar) [reply jobs/pipelines/powervm/zstreams/zstream-trigger-job/Jenkinsfile:5]: ok I will remove it

**Review by Shilpi-Das1 — APPROVED** [review]
/lgtm

File: jobs/pipelines/powervm/zstreams/zstream-ocp4x-p8-min-current-upgrade/Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+
+        //Env constants
+        HARDWARE_CHOSE = "P8"
+        AVAILABILITY_ZONE = "zstream"
+        TARGET = "deploy-openshift4-powervc"
+        TEMPLATE_FILE = ".${TARGET}.tfvars.template"
+        POWERVS = false
+        SCRIPT_DEPLOYMENT = false
+        WAIT_FOR_DEBUG = "0"
+        REDHAT_RELEASE = "8.3"
```
Comment (sudeeshjohn) [inline jobs/pipelines/powervm/zstreams/zstream-ocp4x-p8-min-current-upgrade/Jenkinsfile:48]: lets use rhel8.4

File: jobs/pipelines/powervm/zstreams/zstream-ocp4x-p8-min-current-upgrade/Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+                    wrap([$class: 'BuildUser']) {
+                        env.INSTANCE_NAME = "rdr-zscurst"
+                    }
+                    env.TERRAFORM_VER = "0.13.0"
+                    env.OS_TENANT_NAME="icp-test"
+                    env.OS_NETWORK = "workload"
+                    env.OS_PRIVATE_NETWORK = "workload"
+                    env.SCG_ID = "df21cec9-c244-4d3d-b927-df1518672e87"
+                    env.MOUNT_ETCD_RAMDISK = "false"
+                    env.VOLUME_STORAGE_TEMPLATE = "V7K base template"
```
Comment (sudeeshjohn) [inline jobs/pipelines/powervm/zstreams/zstream-ocp4x-p8-min-current-upgrade/Jenkinsfile:72]: lets run this on CI PowerVC

File: jobs/pipelines/powervm/zstreams/zstream-ocp4x-p8-min-current-upgrade/Jenkinsfile (hunk 4 above)
Comment (varad-ahirwadkar) [reply jobs/pipelines/powervm/zstreams/zstream-ocp4x-p8-min-current-upgrade/Jenkinsfile:48]: ok

**Review by sudeeshjohn — APPROVED** [review]
/lgtm

**Pull request merged**